java/main/src/main/java/com/saucelabs/saucebindings/SauceSession.java

```diff
@@ -118,6 +121,121 @@ public void stop(Boolean passed) {
+        System.out.println("\nYou can take manual control of the test from the Sauce Labs UI here: " + sauceTestLink);
+        this.driver = null;
+    }
+
+    /**
+     * Turns off logging of commands within the test in order to omit sensitive data from the log.json file
+     * NOTE: This method does not omit the commands from other records.
+     * @see #enableLogging()
+     * @see <a href="https://docs.saucelabs.com/basics/test-config-annotation/test-annotation/#methods">Test Annotation Methods</a>
+     */
+    public void disableLogging() {
+        validateSessionStarted("disableLogging()");
+        driver.executeScript("sauce: disable log");
+    }
+
+    /**
+     * Turns logging back on after logging has been disabled
+     * @see #disableLogging()
+     * @see <a href="https://docs.saucelabs.com/basics/test-config-annotation/test-annotation/#methods">Test Annotation Methods</a>
+     */
+    public void enableLogging() {
+        validateSessionStarted("enableLogging()");
+        driver.executeScript("sauce: enable log");
+    }
+
+    /**
+     * Stops VM’s network connection.
+     * @see #startNetwork()
+     * @see <a href="https://docs.saucelabs.com/basics/test-config-annotation/test-annotation/#methods">Test Annotation Methods</a>
+     */
+    public void stopNetwork() {
+        validateSessionStarted("stopNetwork()");
+        SaucePlatform platformName = sauceOptions.getPlatformName();
+        if (!platformName.isMac()) {
+            throw new InvalidArgumentException("Can only stop network for a Mac Platform; current platform is: " + platformName);
+        }
+
+        driver.executeScript("sauce: stop network");
+    }
+
+    /**
+     * Restarts the VM’s network connection after it has been stopped
+     * @see #stopNetwork()
+     * @see <a href="https://docs.saucelabs.com/basics/test-config-annotation/test-annotation/#methods">Test Annotation Methods</a>
+     */
+    public void startNetwork() {
+        validateSessionStarted("startNetwork()");
+        SaucePlatform platformName = sauceOptions.getPlatformName();
+        if (!platformName.isMac()) {
+            throw new InvalidArgumentException("Can only stop network for a Mac Platform; current platform is: " + platformName);
+        }
+
+        driver.executeScript("sauce: start network");
+    }
+
+    /**
+     * This is typically set in SauceOptions before starting the Session.
+     * It might be useful to set dynamically in some situations (e.g., making failing tests more visible
+     *
+     * TODO: Implement this with API because this is not currently working
+     *
+     * @param visibility who can see the job details in the Sauce Labs UI
+     * @see <a href="https://docs.saucelabs.com/basics/test-config-annotation/test-annotation/#methods">Test Annotation Methods</a>
+     * @see BaseConfigurations#setJobVisibility(JobVisibility)
+     */
+    public void setVisibility(JobVisibility visibility) {
```

Comment:
I never liked the term "job", it's a `test`. Can it be `setTestVisibility()`?

Reply:
done